tests/e2e/specs/modules/analytics/dashboard-date-range.test.js

```diff
@@ -33,30 +34,36 @@ describe( 'date range filtering on dashboard views', () => {
 
 		await page.setRequestInterception( true );
 		useRequestInterception( ( request ) => {
-			if ( request.url().match( 'google-site-kit/v1/data/' ) ) {
+			const url = request.url();
+			if ( url.match( 'google-site-kit/v1/modules/analytics' ) ) {
+				let response;
+
+				if ( url.match( 'notifications' ) ) {
+					response = [];
+				} else {
+					const query = getQueryArgs( url );
+					response = getAnalyticsMockResponse( query, false );
```

Comment:
Per [your comment above](https://github.com/google/site-kit-wp/pull/3672#issuecomment-877646962) we shouldn't change the handling of the seed to get a different response. I'm not sure why it would return the same response for different report queries anyways, but even if it did, I don't think it should affect this test, does it?

I see what the problem is now and it seems to me that it might be a problem with the mock response generation. I think it's from this part where we always take the minimum, maximum, and total values from a fixed place in the results which would always be the same due to faker generating the same value in that iteration 🤔  https://github.com/google/site-kit-wp/blob/d54b6af68bfc943dd5b2662e7c94032efef589d7/assets/js/modules/analytics/util/data-mock.js#L274-L280
I think we need to fix this to calculate the values, not necessarily for accuracy, but we do reference these in various places such as this so it is important that it functions a bit more realistically. cc: @eugene-manuilov

Reply:
> I'm not sure why it would return the same response for different report queries anyways, but even if it did, I don't think it should affect this test, does it?

This test depends on there being a different response when the date range is changed. 

> I see what the problem is now and it seems to me that it might be a problem with the mock response generation. 

Aha, well spotted. How should I fix this - assign a random key for the minimum, maximum, and total values? Apply some kind of `sort`?

Happy to do as advised here, let me know what you think.

Reply:
> Aha, well spotted. How should I fix this - assign a random key for the minimum, maximum, and total values? Apply some kind of sort?

@johnPhillips, just loop through rows and find minimums, maximums and count totals for every metric.

Reply:
Just a note on this. After a discussion with @aaemnnosttv we decided that this refactor of `getAnalyticsMockResponse` should probably be tackled in a separate issue. For now I've made the tests pass by using a different `url` field.